flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/operators/AbstractAsyncStateStreamOperator.java

```diff
@@ -300,32 +312,36 @@ public <K, N> InternalTimerService<N> getInternalTimerService(
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public void setKeyContextElement1(StreamRecord record) throws Exception {
+        // This method is invoked only when isAsyncStateProcessingEnabled() is false.
```

Comment:
nit: Add a check here?
Check `isAsyncStateProcessingEnabled = false`

Reply:
I'd suggest not. We are on the hot path of record processing here, it would be nice if we can prevent some overhead here

Reply:
Thanks for the clarification, let's keep it.